mariadb-operator.yaml

```
@@ -0,0 +1,107 @@
+    - uses: test/kwok/cluster
+    - name: Install MariaDB operator CRDs
+      runs: |
+        # Install MariaDB operator CRDs from upstream
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_mariadbs.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_connections.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_backups.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_restores.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_sqljobs.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_databases.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_users.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_grants.yaml
+        kubectl apply -f https://raw.githubusercontent.com/mariadb-operator/mariadb-operator/main/config/crd/bases/k8s.mariadb.com_maxscales.yaml
+        kubectl wait --for=condition=Established crd --all --timeout=60s
+    - name: Launch operator with dummy kubeconfig
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          kubectl config view --minify --raw > /tmp/kubeconfig.yaml
+          # Create webhook certificates
+          mkcert example.com mariadb.local localhost 127.0.0.1 ::1
+          mkdir -p /tmp/k8s-webhook-server/serving-certs
+          mv example.com+4-key.pem /tmp/k8s-webhook-server/serving-certs/tls.key
+          mv example.com+4.pem /tmp/k8s-webhook-server/serving-certs/tls.crt
+          # Create service account directory structure
+          mkdir -p /var/run/secrets/kubernetes.io/serviceaccount
+          echo "default" > /var/run/secrets/kubernetes.io/serviceaccount/namespace
+          echo "dummy-token" > /var/run/secrets/kubernetes.io/serviceaccount/token
+          echo "dummy-ca-cert" > /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+        start: env KUBECONFIG=/tmp/kubeconfig.yaml ${{vars.bin-name}}
+        timeout: 30
+        expected_output: |
+          Watching all namespaces
+          Discovery info
+          Resources
+          Starting manager
+          Starting metrics server
+          Starting EventSource
+          Starting Controller
+          Starting workers
+        post: |
+          echo "Verifying metrics endpoint"
+          curl -sf http://127.0.0.1:8080/metrics | head -20
+          echo "Metrics endpoint is serving as expected"
```

Can we use `grep` instead of `head`? it might be leading false positive.

```suggestion
          curl -sf http://127.0.0.1:8080/metrics | head -20
```